lib/index.js

```diff
@@ -38,6 +42,35 @@ function initialize(config) {
   });
 }
 
+function filterPreviousUploads(uploadListPath, objects) {
+  var uploadedFileList;
```

Comment:
Yeah that would work if we removed the null check on fileName, then the whole method would be about removing the previously uploaded object and not {previously uploaded} + {object with falsey fileName}